manifests/init.pp

```diff
@@ -61,6 +76,6 @@
   }
 
   class { 'docker':
-    * => $default_options[$release_type] + $options
+    * => $default_options[$release_type] + $options + $_socket_group_option
```

Comment:
As a potential compromise between "logic to make things work as expected" and "all configuration options should be exposed" in [the discussion above](https://github.com/simp/pupmod-simp-simp_docker/pull/1/files/2a0e3a245313ffb9ad6ad35c075270121590afb5#r159241517):

If the merge order was switched around to

```
* => $default_options[$release_type] + $_socket_group_option  + $options`
```
then the parameter-provided `$options` (now on the far right) would always override the left-side elements in the [hash merge](https://puppet.com/docs/puppet/4.8/lang_expressions.html#merging).  This would permit users who really feel like overriding the sensible default logic to do so.

Reply:
@op-ct I'm OK with this suggestion, would you mind updating the code to reflect this change? It will simplify the logic and make it "work as expected" in my opinion.